bee-crypto/src/ternary/bigint/i384/mod.rs

```diff
@@ -0,0 +1,893 @@
+
+            // Case 3: both numbers are negative, but equal.
+            Some((NEGBIT..=UMAX, NEGBIT..=UMAX)) => true,
+
+            // Case 4: only s is negative.
+            Some((NEGBIT..=UMAX, _)) => return Some(Less),
+
+            // Case 5: only o is negative.
+            Some((_, NEGBIT..=UMAX)) => return Some(Greater),
+
+            // Case 6: both are positive, s is greater.
+            Some((s, o)) if s > o => return Some(Greater),
+
+            // Case 7: both are positive, s is less.
+            Some((s, o)) if s < o => return Some(Less),
+
+            // Fallthrough case; only happens if s == o.
+            Some(_) => false,
+
+            // The array inside `I384` always has a length larger zero, so the first element is guaranteed to exist.
+            None => unreachable!(),
+        };
+
+        // Create two separate loops as to avoid repeatedly checking `numbers_negative`.
+        if numbers_negative {
+            for (s, o) in zipped_iter {
+                if s > o {
+                    return Some(Less);
+                } else if s < o {
+                    return Some(Greater);
+                }
+            }
+        } else {
+            for (s, o) in zipped_iter {
+                if s > o {
+                    return Some(Greater);
+                } else if s < o {
+                    return Some(Less);
+                }
+            }
+        }
+
+        Some(Equal)
+    }
+}
+
+impl Ord for I384<BigEndian, U8Repr> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.partial_cmp(other) {
+            Some(ordering) => ordering,
+
+            // The ordering is total, hence `partial_cmp` will never return `None`.
+            None => unreachable!(),
+        }
+    }
+}
+
+impl From<T242<Btrit>> for I384<BigEndian, U8Repr> {
+    fn from(value: T242<Btrit>) -> Self {
+        let i384_le: I384<LittleEndian, U32Repr> = value.into();
+        let i384_be: I384<BigEndian, U32Repr> = i384_le.into();
+        i384_be.into()
+    }
+}
+
+impl TryFrom<T243<Btrit>> for I384<BigEndian, U8Repr> {
+    type Error = Error;
+
+    fn try_from(value: T243<Btrit>) -> Result<Self, Error> {
+        let i384_le: I384<LittleEndian, U32Repr> = value.try_into()?;
+        let i384_be: I384<BigEndian, U32Repr> = i384_le.into();
+        Ok(i384_be.into())
+    }
+}
+
+impl I384<BigEndian, U32Repr> {
+    /// Adds `other` onto `self` in place.
+    pub fn add_inplace(&mut self, other: Self) {
+        let mut overflown = false;
+        let self_iter = self.inner.iter_mut().rev();
+        let other_iter = other.inner.iter().rev();
+
+        for (s, o) in self_iter.zip(other_iter) {
+            let (sum, still_overflown) = s.overflowing_add_with_carry(*o, overflown as u32);
+            *s = sum;
+            overflown = still_overflown;
+        }
+    }
+
+    /// Adds `other` in place, returning the number of digits required accomodate `other` (starting from the least
```

Comment:
There is probably a word missing. I don't understand the "accommodate" part. Please check!